### ISSUE TITLE
Zarf publishes empty packages to private repo when package name is not specified

### ISSUE THREAD

**Issue by Ansible-man** [zarf-dev/zarf#2702]
### Environment
Device and OS: RHEL9
App version: 0.35.0
Kubernetes distro being used: RKE2
Other:

### Steps to reproduce
1. Build a zarf package
2. Publish a package to a private repo without specifying the package name- zarf package publish /path/to/my/package/ oci://my-private-registry/zarfrepo --insecure
3. Attempt to pull the package down with zarf package pull oci://my/private/registry/zarfrepo/packagename:tag --insecure
4.  After seeing that fails check the size in the repo
5. Publish package by specifying the package name - zarf package publish /path/to/my/package/my-package.tar.zst oci://my-private-registry/zarfrepo/ --insecure
6. Attempt to clone and it will work

### Expected result
If only one package exists in the directory zarf detects it and pushes it to the private registry successfully 

### Actual Result
zarf acts like it published the package with the correct name and information but it is a shell with nothing in it that you cannot pull

### Visual Proof (screenshots, videos, text, etc)
unable 
pushing without the package name the size in the registry is 6Kibs
pushing it when specifying the package name the size is 395Mibs

### Severity/Priority
medium

### Additional Context
Add any other context or screenshots about the technical debt here.

**Comment by phillebaba** [comment]
Thank you for the report :rose:

I have reproduced this by pushing to a local instance of Zot.

**Comment by phillebaba** [comment]
This may be some confusion in what publish does does for different source types. When you pass a directory as the argument you are pushing up a skeleton package which does not contain any of the package resources.

> Skeleton packages have not been run through the zarf package create process yet, and thus do not have any remote resources included (no images, repos, or remote manifests and files) thereby retaining any [create-time package configuration templates](https://docs.zarf.dev/ref/values) as they were defined in the original zarf.yaml (i.e. untemplated).

https://docs.zarf.dev/faq/#what-is-a-skeleton-zarf-package

I do understand that it can be confusing as the behavior depends on the input to the command. While the CLI output gives some indication that you have created a skeleton package I think that it could be a bit more clear.

**Comment by Ansible-man** [comment]
Would it a good idea to add a flag to allow zarf publish to publish a zarf packages found in a directory when a directory is passed as the argument? If not I see no reason not to close this issue. Thank you for the response.

**Comment by AustinAbro321** [comment]
Yeah I agree skeleton packages make the UX confusing. I have this issue to track #3317
